src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -519,7 +535,8 @@ public bool IsComparableTo(ConnectionDetails other)
             if (ServerName != other.ServerName
                 || AuthenticationType != other.AuthenticationType
                 || UserName != other.UserName
-                || AzureAccountToken != other.AzureAccountToken)
+                || AzureAccountToken != other.AzureAccountToken
+                || ColumnEncryptionSetting != other.ColumnEncryptionSetting)
```

Comment:
if this is here then other settings should be here too. I think compare only compares connection details and authentication properties

Reply:
When this setting is not present in the `IsComparableTo()` method, I found that adjusting this setting in the Change Connection dialog in ADS did not change anything. That is, if I set an column encryption enabled connection to column encryption disabled then the connection stays column encryption enabled. It appears that when a user attempts to change the connection in this way, then this method determines whether anything has really changed and if so, changes the connection, otherwise, it is like a NOOP.
Once I added this setting to the method then things started working as expected.

Reply:
this looks like a bug to me. we have to check all the settings here then. the scenario is when user connected to a db and wants to change the connection in the dialog. changing settings don't make the connection to reset. we should fix that here and check all the settings.

---
In reply to: [366024645](https://github.com/microsoft/sqltoolsservice/pull/908#discussion_r366024645) [](ancestors = 366024645)

Reply:
This is likely to have undesirable side-effects, since our principle has been the connection identity only contain fields which are visible to the user in the Servers viewlet.  Otherwise, difference connections with different properties will appear identical to the user and this will cause confusion.  If we want to add new properties to the connection identity then we need to also do the work to fix the Server UX to display these properties when there is a possibility of ambiguity.

I would recommend to remove this change from this PR, and open an issue for the change connection bug (which likely is not specific to Column Encryption).  We can discuss how to fix the issue for all properties, but that should be a largely update which address the related issue holistically.

Reply:
I removed this change and submitted the [change connection bug](https://github.com/microsoft/sqltoolsservice/issues/913) as recommended.